wgpu/src/triangle.rs

```diff
@@ -261,65 +272,85 @@ impl Pipeline {
+                            indices_size,
+                            device,
+                        );
+
+                        index_buffer.copy_from_slice(indices);
+                    }
+
+                    uniforms.push(transform);
+                    offsets.push((
+                        last_vertex as u64,
+                        last_index as u64,
+                        mesh.buffers.indices.len(),
+                    ));
+
+                    last_vertex += mesh.buffers.vertices.len();
+                    last_index += mesh.buffers.indices.len();
+                }
+            }
+        }
 
-            encoder.copy_buffer_to_buffer(
-                &vertex_buffer,
-                0,
-                &self.vertex_buffer.raw,
-                (std::mem::size_of::<Vertex2D>() * last_vertex) as u64,
-                (std::mem::size_of::<Vertex2D>() * mesh.buffers.vertices.len())
-                    as u64,
-            );
+        let uniforms = uniforms.as_bytes();
 
-            encoder.copy_buffer_to_buffer(
-                &index_buffer,
+        if let Some(uniforms_size) =
+            wgpu::BufferSize::new(uniforms.len() as u64)
+        {
+            let mut uniforms_buffer = staging_belt.write_buffer(
+                encoder,
+                &self.uniforms_buffer.raw,
                 0,
-                &self.index_buffer.raw,
-                (std::mem::size_of::<u32>() * last_index) as u64,
-                (std::mem::size_of::<u32>() * mesh.buffers.indices.len())
-                    as u64,
+                uniforms_size,
+                device,
             );
 
-            uniforms.push(transform);
-            offsets.push((
-                last_vertex as u64,
-                last_index as u64,
-                mesh.buffers.indices.len(),
-            ));
-
-            last_vertex += mesh.buffers.vertices.len();
-            last_index += mesh.buffers.indices.len();
+            uniforms_buffer.copy_from_slice(uniforms);
         }
 
-        let uniforms_buffer = device.create_buffer_with_data(
-            uniforms.as_bytes(),
-            wgpu::BufferUsage::COPY_SRC,
-        );
-
-        encoder.copy_buffer_to_buffer(
-            &uniforms_buffer,
-            0,
-            &self.uniforms_buffer.raw,
-            0,
-            (std::mem::size_of::<Uniforms>() * uniforms.len()) as u64,
-        );
-
         {
-            let (attachment, resolve_target, load_op) =
+            let (attachment, resolve_target, load) =
                 if let Some(blit) = &mut self.blit {
                     let (attachment, resolve_target) =
                         blit.targets(device, target_width, target_height);
 
-                    (attachment, Some(resolve_target), wgpu::LoadOp::Clear)
+                    (
+                        attachment,
+                        Some(resolve_target),
+                        wgpu::LoadOp::Clear(wgpu::Color {
```

Comment:
`wgpu::Color::TRANSPARENT`

Reply:
Done in 8d605be4e3e0b678c7563708ea941d6188f45678.